File: 242. Valid Anagram/Solution2.java
public class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length()) {
            return false;
        }

        int[] sum = new int[26];
        for(int i = 0; i < s.length(); ++i) {
            sum[s.charAt(i)-'a']++;
        }
        for(int j = 0; j < t.length(); ++j) {
            --sum[t.charAt(j)-'a'];
            if(sum[t.charAt(j)-'a'] < 0) {
                return false;
            }
        }
        
        return true;
    }
}

File: Algorithms/274. H-Index/solution2.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        N = len(citations)
        cnts = [0] *(N+1)
        for c in citations:
            cnts[[c,N][c>N]] += 1
        sum = 0
        for h in range(N,0,-1):
            if sum + cnts[h] >= h:
                return h
            sum += cnts[h]
        return 0